src/Components/Web/src/DataTransfer.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// The <see cref="DataTransfer"/> object is used to hold the data that is being dragged during a drag and drop operation.
+    /// It may hold one or more <see cref="UIDataTransferItem"/>, each of one or more data types.
+    /// For more information about drag and drop, see HTML Drag and Drop API.
+    /// </summary>
+    public class DataTransfer
```

Comment:
This seems a bit tangential. Is the behavior complex enough to warrant any test cases, or do you see it as just a DTO?

I'm unsure whether all the properties here would actually be usable to achieve some useful result. For example at looks like `EffectAllowed` has to be set synchronously during a `dragstart` event, which wouldn't be possible because:

1. In server-side Blazor, event handling is async
2. Event in client-side Blazor, the event args are a .NET copy, not the original JS object. Mutating the .NET copy doesn't mutate the original JS object.

If this is correct, perhaps we could reduce the set of properties to those that can do something useful to avoid confusion.

Reply:
Actually this is an example of a broader question, so resolving this individual subthread.